### PR TITLE
Added recipe for girder

### PR DESCRIPTION
**Pull request by pmlandwehr**
(Pinging a maintainer out of thread)

### CODE REVIEW

**Comment by pmlandwehr** [comment]
Crash out during the tests due to `from pkg_resources import iter_entry_points`. Wondering if this is an artifact and we don't need `setuptools` at runtime, but I've pushed it in.

**Comment by ocefpaf** [comment]
> @ocefpaf is there a best practice for handling extras and their requirements? Build nthem all?

That is a good question. There is no standard yet and I welcome your opinion here. This is what I do:

- if I need it I add it
- if it is a lightweight package I add
- if it is "heavy" I don't add

File: recipes/girder/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+requirements:
+  build:
+    - python
+    - setuptools
+
+  run:
+    - python
+    - pip
+    - setuptools
```
Comment (ocefpaf) [inline recipes/girder/meta.yaml]: `pip` and `setuptools` at run time?

**Comment by pmlandwehr** [comment]
I've often been including them and/or building them as extra repos. But it currently seems like conda doesn't allow for selecting extras at build time - e.g `girder[plugins]` Is there a workaround for this?

**Comment by ocefpaf** [comment]
> I've often been including them and/or building them as extra repos. But it currently seems like conda doesn't allow for selecting extras at build time - e.g `girder[plugins]` Is there a workaround for this?

You can create meta-packages, but I don't really like them. I prefer to ship a "bulky" package with all the extras/plugins, etc.

**Comment by pmlandwehr** [comment]
@ocefpaf sounds good. In that case this needs some packages that will cause breaks, e.g. #1043

**Comment by ocefpaf** [comment]
>  In that case this needs some packages that will cause breaks

Why #1043 will cause breaks here?

**Comment by pmlandwehr** [comment]
Rather, without #1043 a fuller build of this (yet to be uploaded, will fail - `girder` extras depend on various `hachoir-*` modules.

**Comment by ocefpaf** [comment]
Got it. I just merged #1043 BTW

File: recipes/girder/meta.yaml (hunk 1 above)
Comment (pmlandwehr) [inline recipes/girder/meta.yaml]: Yeah - I was surprised too, but definitely.

Comment (ocefpaf) [inline recipes/girder/meta.yaml]: If the metadata says so that is it :smile_cat: 

Ready to merge?

**Comment by pmlandwehr** [comment]
(Close/Reopen now that all `hachoir-*` components are done)

**Comment by pmlandwehr** [comment]
[`hachoir-parser`](https://anaconda.org/conda-forge/hachoir-parser) and [`hachoir-metadata`](https://anaconda.org/conda-forge/hachoir-metadata) are both here and in the linux channel, so I'm a little stumped why Circle-CI is breaking. One more open/close...

**Comment by pmlandwehr** [comment]
Bah, python 2.7 for windows is breaking because we don't have `snakebite`. I've played around with building it but wouldn't be able to build `snakebite[kerberos]` because of the dependency on `sasl` (see #842 ) and on `python-krbV`, which I've tried (and failed) to build.) As is, I'm inclined to skip the windows build.

**Comment by ocefpaf** [comment]
> As is, I'm inclined to skip the windows build.

That is fine. Just remember to open an issue in the feedstock later to keep track of it.

**Comment by jakirkham** [comment]
LGTM

**Pull request merged**